sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1135,22 +1137,20 @@ def __init__(self, loss="squared_loss", *, penalty="l2", alpha=0.0001,
 
     def _partial_fit(self, X, y, alpha, C, loss, learning_rate,
                      max_iter, sample_weight, coef_init, intercept_init):
+        first_call = getattr(self, "coef_", None) is None
```

Comment:
Shouldn't this be `hasattr`? Or can coef_ actually exist *and* be None?

Reply:
`coef_` can actually exist and be `None`:

https://github.com/scikit-learn/scikit-learn/blob/cd673475bde70e87255ccd9b6f35687ce59b4b67/sklearn/linear_model/_stochastic_gradient.py#L1205-L1213